apps/frontend/src/context/MockContext.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+declare global {
+  interface Window {
+    setMock: (b?: boolean) => void;
+    mockHr: number | null;
+    mockWatt: number | null;
+  }
+}
+
+const mockingOnFromEnvironment =
+  import.meta.env.VITE_USE_MOCK_DATA === 'true' ||
+  import.meta.env.VITE_USE_MOCK_DATA === 'TRUE' ||
+  false;
+
+const MockContext = React.createContext<{
+  mocking: boolean;
+} | null>({
+  mocking: mockingOnFromEnvironment,
+});
+
+export const MockProvider = ({ children }: { children: React.ReactNode }) => {
+  const [mocking, setMocking] = React.useState(false);
+  React.useEffect(() => {
+    window.setMock = (b: any) => {
+      if (b === undefined) {
+        setMocking(true);
+      }
+      if (typeof b === 'boolean') {
+        setMocking(b);
+      }
+    };
```

Comment:
Should the argument be something like `value?: boolean`? Or perhaps a more strict `value: boolean`? 

If the reason for the undefined typing is to be able to call `setMock` in the browser, perhaps `toggleMock` without any parameters would be easier?

Reply:
Fixed 🚀